# 👋 Hi, I'm Honoka

I'm a bioinformatics masters student at Georgia Institute of Technology 🐝 interested in immunology, gene expression analysis, and single-cell technologies. This GitHub profile hosts a variety of projects I've developed during my academic research.

---

## 🧭 Recent Large Projects

- [**DN2_Atlas**](https://github.com/hmiura21/DN2_Atlas)  
  Single-cell atlas of DN2 B cells revealing novel subtypes across diseases and demographics.

- [**NPI_GeneExpression_Prediction**](https://github.com/hmiura21/NPI_GeneExpression_Prediction)  
  Predicting Nottingham Prognostic Index from gene expression using machine learning.

- [**Bioinformatics_Magnumopus**](https://github.com/hmiura21/bioinformatics_magnumopus)  
  A custom pipeline to detect 16S amplicons from reads using isPCR and metadata.

- [**Airline Management System**](https://github.com/hmiura21/AirlineManagementSystem)  
  Airline reservation and management tool built with SQL, supporting passenger bookings and flight queries.

---


## 🌱 Tools/ Skills I am Currently Learning

- Command-line bioinformatics tools such as BLAST, Prodigal, Barrnap
- Data Visualization in R
- Statistical Analysis using R
- Machine Learning Pipelines using PyTorch (ESM2, ProtT5)
- Single-cell RNA analysis using R tools (Seurat)
- Querying and Creating SQL tables
- Developing interactive web applications using Shiny
- Custom CLI pipeline
- Workflow managemnt system such as Nextflow
- Quality control, dimentionality reduction using R and Python
- Clustering analysis and differential gene expression using R
- Managing large datasets (.rds, .csv, .h5ad, .fa, .bed)

---


## 🧭 Other Past Projects 

- [**Protein Mutation Prediction**](https://github.com/hmiura21/ProteinMutationPrediction)  
  Machine Learning model to predict the functional effects of single-point protein mutations using ESM2 and parameter-efficient fine-tuning via LoRA

- [**QC Workflow**](https://github.com/hmiura21/QC_Workflow)  
  Nextflow pipeline that performs sequential read quality control and genome assembly while also analyzing the quality-filtered reads in parallel.

- [**Extract Homolog Gene**](https://github.com/hmiura21/Extract_Homolog_Gene)  
   Enhanced version of the find_homologs.py script that identifies putative homologous genes from BLAST results, maps them to gene annotations in a BED file, extracts their sequences from a reference assembly, and writes the results to a     FASTA file

- [**Align Sequences**](https://github.com/hmiura21/AlignSequences)  
  Python script that visualizes the alignment of two DNA sequences from a FASTA file

- [**Homolog Identify**](https://github.com/hmiura21/Homolog_Identify)  
  Bash script designed to identify putative histidine kinase (HK) homologs in a nucleotide genome and determine which genes (from a provided BED file) contain those homologs.

- [**Find Perfect Matches**](https://github.com/hmiura21/Find_Perfect_Matches)  
  Bash script to identify perfect matches between a query FASTA file and a subject FASTA file using BLAST

- [**Comparative Genomics**](https://github.com/hmiura21/ComparativeGenomics)  
  Bash script to investigate a potential outbreak by analyzing 10 Illumina-sequenced bacterial isolates from NCBI's SRA database

- [**Genotyping**](https://github.com/hmiura21/Genotyping)  
  Genomic analysis of Bordetella parapertussis using FastANI, MLST, and CheckM

- [**Gene Prediction and Annotation**](https://github.com/hmiura21/GenePredictionAndAnnotation)  
  Bash script to annotate a bacterial genome and extract key sequences using Prodigal, Barrnap, and BLAST

- [**QC Assembly**](https://github.com/hmiura21/QC_Assembly)  
  Bash script to perform read trimming and genome assembly using Fastp and Skesa for 34 files of unknown bacterial samples.

- [**Global Temperatures Visualization**](https://github.com/hmiura21/GlobalTemperaturesVisualization)  
  Python analysis of global average temperature anomalies from January 1850 to the present using monthly data obtained from the NOAA


---


## 📫 Get in Touch

- 🧑‍🎓 Graduate Student at Georgia Institute of Technology
- 📧 Email: konoha1031rila@gmail.com
- 🌐 [LinkedIn](www.linkedin.com/in/honoka-miura-279b131b5)

---

> 📝 This profile is actively maintained. Some data used in projects may be restricted for privacy reasons and replaced with examples.

